ivy/functional/backends/numpy/elementwise.py

```diff
@@ -15,6 +15,9 @@
 def add(x1: np.ndarray, x2: np.ndarray) -> np.ndarray:
     if not isinstance(x2, np.ndarray):
         x2 = np.asarray(x2, dtype=x1.dtype)
+    # numpy.array_api doesn't support float16
+    if x1.dtype == "float16" or x2.dtype == "float16":
+        return np.asarray(np.add(np.asarray(x1), np.asarray(x2)))
```

Comment:
we should just return this version always, also, no need for the `np.asarray` in this case. Also I think this function should be symmetric with regards to `x1` and `x2`

Reply:
So we don't use numpy.array_api, ok. Regarding the symmetricity, yes the types are symmetric for most of the cases but for a few test cases, it happened that the dtype of x2 and x1 was different. However, if we are not going to use the numpy.array_api space at all then no need for this line.

Reply:
little more clarification is needed, see my latest commit.

Reply:
yeah, we shouldn't use `np.array_api` at all.